app/src/main/cpp/skyline/services/socket/sfdnsres/IResolver.cpp

```diff
@@ -2,7 +2,176 @@
+        response.Push<u32>(dataSize);
+        return {};
+    }
+
+    Result IResolver::GetHostByNameRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return {};
+    }
+
+    Result IResolver::GetAddrInfoRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto [dataSize, resultCode]{GetAddrInfoRequestImpl(request)};
+        response.Push<i32>(resultCode);  // errno
+        response.Push(AddrInfoErrorToNetDbError(resultCode));  // NetDBErrorCode
+        response.Push<u32>(dataSize);
+        response.Push<u32>(0);
+        return {};
+    }
+
+    Result IResolver::GetNameInfoRequestWithOptions(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return {};
+    }
+
+    std::pair<u32, i32> IResolver::GetAddrInfoRequestImpl(ipc::IpcRequest &request) {
+        auto hostname{request.inputBuf.at(0).as_string(true)};
+        auto service{request.inputBuf.at(1).as_string(true)};
+
+        if (!(*state.settings->isInternetEnabled)) {
+            Logger::Info("Internet access disabled, DNS Blocked: {}", hostname);
+            return {0, -1};
+        }
+
+        addrinfo *result;
+        i32 resultCode = getaddrinfo(hostname.data(), service.data(), nullptr, &result);
+
+        u32 dataSize{0};
+        if (resultCode == 0 && result != nullptr) {
+            const std::vector<u8> data = SerializeAddrInfo(result, resultCode, hostname);
+            dataSize = static_cast<u32>(data.size());
+            request.outputBuf.at(0).copy_from(data);
+            freeaddrinfo(result);
+        }
+        return {dataSize, resultCode};
+    }
+
+    // https://github.com/yuzu-emu/yuzu/blob/ce8f4da63834be0179d98a7720dee47d65f3ec06/src/core/hle/service/sockets/sfdnsres.cpp#L76
+    std::vector<u8> IResolver::SerializeAddrInfo(const addrinfo* addrinfo, i32 result_code, std::string_view host) {
+        std::vector<u8> data;
+
+        auto* current{addrinfo};
+        while (current != nullptr) {
+            struct SerializedResponseHeader {
+                u32 magic;
+                u32 flags;
+                u32 family;
+                u32 socketType;
+                u32 protocol;
+                u32 addressLength;
+            };
+            static_assert(sizeof(SerializedResponseHeader) == 0x18);
+
+            constexpr auto headerSize{sizeof(SerializedResponseHeader)};
+            const auto addrSize{current->ai_addr && current->ai_addrlen > 0 ? current->ai_addrlen : 4};
+            const auto canonnameSize{current->ai_canonname ? strlen(current->ai_canonname) + 1 : 1};
+
+            const auto lastSize{data.size()};
+            data.resize(lastSize + headerSize + addrSize + canonnameSize);
+
+            // Header in network byte order
+            SerializedResponseHeader header{};
+
+            constexpr auto HEADER_MAGIC{0xBEEFCAFE};
```

Comment:
move this to be defined in the struct ctor

      struct SerializedResponseHeader {
                u32 magic{util::SwapEndianness(0xbeefcafe);
                u32 flags;
                u32 family;
                u32 socketType;
                u32 protocol;
                u32 addressLength;
            };